### DISCUSSION TITLE
Not executing the code

### DISCUSSION THREAD

**Discussion by fatemehmomeni80** [optuna/optuna#5293] [Q&A]
### Don't use GitHub Issues to ask support questions.

Hello, I want to find the most optimal hyperparameters of yolo v8 and I tried this code, but it can't be trained, why?

```
import optuna
import subprocess
import re

def objective(trial):
    learning_rate = trial.suggest_float('learning_rate', 1e-5, 1e-1, log=True)
    batch_size = trial.suggest_categorical('batch_size', [4, 8, 16, 32])
    weight_decay = trial.suggest_float('weight_decay', 1e-6, 1e-3, log=True)

    train_command = [
        "yolo", 
        "task=detect", 
        "mode=train", 
        "model=yolov8s.pt",  
        "data=/content/yolov8/datasets/ships.yaml", 
        "epochs=20", 
        f"imgsz=640",  
        f"learning_rate={learning_rate}", 
        f"batch_size={batch_size}", 
        f"weight_decay={weight_decay}"
    ]

    try:
        result = subprocess.run(train_command, capture_output=True, text=True, timeout=1800)  
    except subprocess.TimeoutExpired:
        return 0.0

    map_value = parse_map_from_output(result.stdout)

    return map_value

def parse_map_from_output(output):
    map_pattern = re.compile(r"mAP:\s*([\d.]+)%")
    match = map_pattern.search(output)
    
    if match:
        map_value = float(match.group(1))
        return map_value
    else:
        return 0.0

study = optuna.create_study(direction='maximize')  
study.optimize(objective, n_trials=50)

best_hyperparameters = study.best_params
print(f" best hyperparametr: {best_hyperparameters}"

```

output:

[I 2024-03-04 12:08:01,171] A new study created in memory with name: no-name-3de5d4b9-0538-4886-b627-2173c0b178fa
[I 2024-03-04 12:08:06,082] Trial 0 finished with value: 0.0 and parameters: {'learning_rate': 0.01747455422682767, 'batch_size': 4, 'weight_decay': 6.894327052970726e-06}. Best is trial 0 with value: 0.0.
[I 2024-03-04 12:08:10,332] Trial 1 finished with value: 0.0 and parameters: {'learning_rate': 0.010286502463242897, 'batch_size': 8, 'weight_decay': 8.483322725129547e-06}. Best is trial 0 with value: 0.0.
[I 2024-03-04 12:08:14,966] Trial 2 finished with value: 0.0 and parameters: {'learning_rate': 0.0029346641716051485, 'batch_size': 16, 'weight_decay': 2.3465754198921045e-05}. Best is trial 0 with value: 0.0.

why??

**Comment by nzw0301** [comment]
It think each training takes more than timeout, 1800 seconds.  Could you remove the timeout?
